src/applications/personalization/profile/actions/directDeposit.js

```diff
@@ -98,16 +98,56 @@ export function saveDirectDeposit(
 
     const response = await getData(DIRECT_DEPOSIT_API_ENDPOINT, {
       method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+      },
       body: JSON.stringify(data),
     });
 
-    // error handling for the response
-    if (response.error || response.errors) {
-      recordDirectDepositEvent({
+    // for client side errors
+    // check if response is an instance of an error TypeError or any other error
+    if (response instanceof Error) {
+      const event = {
+        method: 'PUT',
         endpoint: DIRECT_DEPOSIT_API_ENDPOINT,
         status: API_STATUS.FAILED,
+        extraProperties: {
+          'error-key':
+            response?.message || 'unknown-error error instance message',
+        },
+      };
+
+      recordDirectDepositEvent(event);
+
+      captureDirectDepositError(response.message, {
+        eventName: 'put-direct-deposit-failed',
+      });
+
+      dispatch({
+        type: DIRECT_DEPOSIT_SAVE_FAILED,
+        response,
       });
 
+      return;
+    }
+
+    // error handling for the response if it is a server side error
+    if (response.error || response.errors) {
+      const errorStatus =
+        response.errors?.[0]?.status || response?.status || '';
+      const errorCodeOrMessage =
+        response.errors?.[0]?.code || response?.error || 'unknown-error';
```

Comment:
Parsing the error code or the error message depending on if it is an errors array or an error string (500 errors are a string for instance) I also thought tagging on the status code would be useful for GA as well, and that will just be appended to the end of the 'error-key' that we send GA